File: src/DepositfilesSDK/DepositfilesSDK/IClient.vb
Imports DepositfilesSDK.JSON

Public Interface IClient

    ReadOnly Property Files(FileID As String) As IFiles
    ReadOnly Property Folders(FolderID As String) As IFolders
    ReadOnly Property RootID As String
    ReadOnly Property Root As IRoot


    Function UserInfo() As Task(Of JSON_GetToken)

End Interface

File: DepositfilesSDK/DepositfilesSDK/IClient.cs
using System.Threading.Tasks;
using DepositfilesSDK.JSON;

namespace DepositfilesSDK
{
    public interface IClient
    {
        IFile File(string FileID);
        IFiles Files(string[] FilesID);
        IFolder Folder(string FolderID);
        IRoot Root();
        Task<JSON_GetToken> UserInfo();
    }
}